patches/minecraft/net/minecraft/block/BlockRedstoneComparator.java.patch

```diff
@@ -7,7 +7,7 @@
 +    @Override
 +    public void onNeighborChange(IBlockAccess world, BlockPos pos, BlockPos neighbor)
 +    {
-+        if (pos.func_177956_o() == neighbor.func_177956_o() && world instanceof World)
++        if (pos.func_177956_o() == neighbor.func_177956_o() && world instanceof net.minecraft.world.WorldServer)
```

Comment:
Vanilla never triggers `func_189540_a` (i.e. `neighborChanged`) on the client. Forge's `onNeighborChange` however does fire on the client (i.e. when a TE is removed). That causes the client to say "hey, this comparator is no longer valid, lets remove it", without waiting for the server's authority.